mart/attack/callbacks/base.py

```diff
@@ -23,60 +23,60 @@ class Callback(abc.ABC):
 
     def on_run_start(
         self,
-        adversary: Adversary,
-        input: torch.Tensor | tuple,
-        target: torch.Tensor | dict[str, Any] | tuple,
-        model: torch.nn.Module,
+        adversary: Adversary | None = None,
+        input: torch.Tensor | tuple | None = None,
+        target: torch.Tensor | dict[str, Any] | tuple | None = None,
+        model: torch.nn.Module | None = None,
         **kwargs,
     ):
         pass
 
     def on_examine_start(
         self,
-        adversary: Adversary,
-        input: torch.Tensor | tuple,
-        target: torch.Tensor | dict[str, Any] | tuple,
-        model: torch.nn.Module,
+        adversary: Adversary | None = None,
+        input: torch.Tensor | tuple | None = None,
+        target: torch.Tensor | dict[str, Any] | tuple | None = None,
+        model: torch.nn.Module | None = None,
         **kwargs,
     ):
         pass
 
     def on_examine_end(
         self,
-        adversary: Adversary,
-        input: torch.Tensor | tuple,
-        target: torch.Tensor | dict[str, Any] | tuple,
-        model: torch.nn.Module,
+        adversary: Adversary | None = None,
+        input: torch.Tensor | tuple | None = None,
+        target: torch.Tensor | dict[str, Any] | tuple | None = None,
+        model: torch.nn.Module | None = None,
         **kwargs,
     ):
         pass
 
     def on_advance_start(
         self,
-        adversary: Adversary,
-        input: torch.Tensor | tuple,
-        target: torch.Tensor | dict[str, Any] | tuple,
-        model: torch.nn.Module,
+        adversary: Adversary | None = None,
+        input: torch.Tensor | tuple | None = None,
+        target: torch.Tensor | dict[str, Any] | tuple | None = None,
+        model: torch.nn.Module | None = None,
         **kwargs,
     ):
         pass
 
     def on_advance_end(
         self,
-        adversary: Adversary,
-        input: torch.Tensor | tuple,
-        target: torch.Tensor | dict[str, Any] | tuple,
-        model: torch.nn.Module,
+        adversary: Adversary | None = None,
+        input: torch.Tensor | tuple | None = None,
+        target: torch.Tensor | dict[str, Any] | tuple | None = None,
+        model: torch.nn.Module | None = None,
         **kwargs,
     ):
         pass
 
     def on_run_end(
         self,
-        adversary: Adversary,
-        input: torch.Tensor | tuple,
-        target: torch.Tensor | dict[str, Any] | tuple,
-        model: torch.nn.Module,
+        adversary: Adversary | None = None,
+        input: torch.Tensor | tuple | None = None,
+        target: torch.Tensor | dict[str, Any] | tuple | None = None,
+        model: torch.nn.Module | None = None,
```

Comment:
I think you can revert all of these changes since the arguments ought to be required. However, you should make these required keyword arguments.

Reply:
Fixed.